core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -787,6 +811,15 @@ trait EntityComponent {
       }
     }
 
+    def doesEntityTypeAlreadyExist(workspaceContext: Workspace, entityType: String): ReadAction[Seq[Boolean]] = {
+      CheckForExistingEntityTypeQuery.doesEntityTypeAlreadyExist(workspaceContext, entityType)
+    }
+
+    def changeEntityTypeName(workspaceContext: Workspace, oldName: String, newName: String): ReadWriteAction[Int] = {
+      ChangeEntityTypeNameQuery.changeEntityTypeName(workspaceContext, oldName, newName) andThen
+        workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID)
+    }
```

Comment:
by putting `workspaceQuery.updateLastModified` second here, the result of `changeEntityTypeName` will be the number of rows updated by  `workspaceQuery.updateLastModified`, not the number of rows updated by `ChangeEntityTypeNameQuery.changeEntityTypeName`. This seems wrong. One option is to swap the order of calls here, but that changes internal logic. To keep the order of operations the same but return the result of `ChangeEntityTypeNameQuery.changeEntityTypeName`, you could do:

```scala
def changeEntityTypeName(workspaceContext: Workspace, oldName: String, newName: String): ReadWriteAction[Int] = {
  for {
    numRowsRenamed <- ChangeEntityTypeNameQuery.changeEntityTypeName(workspaceContext, oldName, newName)
    _ <- workspaceQuery.updateLastModified(workspaceContext.workspaceIdAsUUID)
  } yield {
    numRowsRenamed
  }
}
```
(for-yield constructs like this result in flatMaps, just like andThen results in a flatMap)

Reply:
this is the only outstanding issue - should `changeEntityTypeName` return the count of rows that had their entity type changed, or should it always return a value of `1` for the count of workspaces that had their last-modified column changed?

Reply:
I've made the change to return the number of rows modified by the update ENTITY sql statement.